specification/under_development/Base-Profile.md

```diff
@@ -0,0 +1,160 @@
+  tail call void @__quantum__rt__tuple_start_record_output()
+  tail call void @__quantum__rt__result_record_output(%Result* null)
+  tail call void @__quantum__rt__result_record_output(%Result* inttoptr (i64 1 to %Result*))
+  tail call void @__quantum__rt__tuple_end_record_output()
+
+  ret void
+}
+
+; declarations of qis functions
+
+declare void @__quantum__qis__h__body(%Qubit*)
+
+declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*)
+
+declare void @__quantum__qis__mz__body(%Qubit*, %Result*)
+
+; declarations of functions used for output recording
+
+declare void @__quantum__rt__tuple_start_record_output()
+
+declare void @__quantum__rt__result_record_output(%Result*)
+
+declare void @__quantum__rt__tuple_end_record_output()
+
+; attributes
+
+attributes #0 = { "EntryPoint" "requiredQubits"="2" "requiredResults"="2" }
+```
+
+TODO: a profile identifier and version number within the IR would be good to have (may be needed for correct usage of qubit and result pointers)   
+-> look into custom target triples
+
+## Entry Point Definition
+
+The bitcode contains the definition of the llvm function that should be invoked when the program is executed, 
+refered to as entry point in the rest of this profile specification. 
+The name of this function may be chosen freely, as long as it is a valid 
+[global identifier](https://llvm.org/docs/LangRef.html#identifiers) by LLVM standard. 
+
+The entry point may not take any parameters and must return void. 
+
+### Attributes
+
+The following custom attributes must be attached to the entry point function:
+
+- An attribute named "EntryPoint" that does not 
+- An attribute indicating the total number of qubits used by the program
+- An attribute indicating the total number of classical bits required throughout the program to store measurement outcomes
+
+These attributes will show up as an [attribute group](https://releases.llvm.org/13.0.1/docs/LangRef.html#attrgrp) in the IR.
+Attribute groups are numbered such that they can be easily referenced by multiple function definitions or global variables. 
+Arbitrary custom attributes may be optionally attached to any of the declared functions to convey additional information about that function. Consumers of Base Profile compliant programs should hence not rely on the numbering of the entry point attribute group, but instead look for function to which an attribute with the name `EntryPoint` is attached to determine which one to invoke when the program is launched.
+
+To indicate the total number of qubits required to execute the program, a custom attribute with the name `requiredQubits` is defined and attached to the entry point. To indicate the number of registers/bits needed to store measurement results during program execution, a custom attribute with the name `requiredResults` is defined and attached to the entry point. The value of both of these attributes is the string representation of a 32-bit integer constant.
```

Comment:
For consistency with the pointer types defined below these should probably be 64-bit integers.